p2p/host.go

```diff
@@ -693,6 +696,22 @@ func (host *HostV2) Connected(net libp2p_network.Network, conn libp2p_network.Co
 	}
 }
 
+// getPeerTopics returns the list of topics a peer is subscribed to
+func (host *HostV2) getPeerTopics(peerID libp2p_peer.ID) map[string]bool {
```

Comment:
that function is currently not used anywhere.  Is this required in this PR ? also please write the unit test for this function

Reply:
This function has been added for potential future use. If we later need to retrieve peer topics, we can use it. I included it preemptively because I believe it might be useful.

Reply:
noted please the add unit test

Reply:
@GheisMohammadi can you add unit test for getPeerTopics or maybe create a seperate PR for that function with its test ? what do you think ?

Reply:
@GheisMohammadi